sdk/cosmos/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/core/CosmosTemplate.java

```diff
@@ -801,40 +792,40 @@ private <T> Slice<T> sliceQuery(SqlQuerySpec querySpec,
         }
 
         final List<T> result = new ArrayList<>();
-        for (int index = 0; it.hasNext()
-            && index < pageable.getPageSize() + offsetForPageWithoutContToken; index++) {
+        while (result.size() < pageable.getPageSize() && it.hasNext()) {
```

Comment:
I feel the logic here can be more simplified if we move the first feedResponse into the loop as well:

```
int  initialPageSize;
string initialContinuationToken;
do {
   //createFeedResponse
  // emit result
 // update continuationToken
// update page size
} while (result.size() < pageable.getPageSize() && continuationToken != null)

```

Reply:
I'm not sure how we would move the first one into the loop, do you have any suggestions? We need the while loop to be dependent on the it.hasNext() but we won't have it if we move the first one into the loop

Reply:
Let me know what you think of my latest commit.